### PR TITLE
[GDR-2498] Move plot functions from gDRcomponents to gDRplots 

### PR DESCRIPTION
**Pull request by j-smola**
# Description
## What changed?
Related JIRA issue: [GDR-2498](https://jira.gene.com/jira/browse/GDR-2498)

Move convert_se_assay_to_custom_dt, capVals and get_settings_from_json from gDRcomponents package

## Why was it changed?

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [x] Package version bumped
- [x] Changelog updated

# Screenshots (optional)

### CODE REVIEW

**Review by gladkia — APPROVED** [review]

File: R/convert_mae_se_assay_to_dt.R [hunk 1]
```diff
@@ -226,3 +226,195 @@ convert_mae_assay_to_dt <- function(mae,
...
+#' Convert an assay within a SummarizedExperiment object to a long data.table. Then
+#' condcut some post processing steps.
+#'
+#' Current strategy is per-assay specific.
+#' 1. combo assays: conversion to data.table only (with `wide_structure` = FALSE)
+#' 2. 'Metrics' asssay can be converted to three types of outputs:
+#'   - Metrics_initial (conversion to data.table only, with `wide_structure` = FALSE)
+#'   - Metrics_raw: same as Metrics_initial followed by:
+#'     * fix for 'EC50' and 'Metrics_rownames'
+#'     * gDRutils::flatten
```
Comment (bczech) [inline R/convert_mae_se_assay_to_dt.R:242]: redundant namespacing

File: tests/testthat/test-convert_mae_se_assay_to_dt.R [hunk 2]
```diff
@@ -197,3 +197,98 @@ test_that("convert_mae_assay_to_dt works as expected", {
   expect_equal(setdiff(colnames(dt_w), col_w), setdiff(colnames(dt_l), col_l))
   expect_equal(max(dt_l[, .N, by = "normalization_type"]$N), NROW(dt_w))
 })
+
+
+test_that("convert_se_assay_to_custom_dt works fine", {
+  json_path <- system.file(package = "gDRcomponents", "settings.json")
+  s <- get_settings_from_json(json_path = json_path)
+  
+  se <- gDRutils::get_synthetic_data("finalMAE_small")[[1]]
```
Comment (bczech) [inline tests/testthat/test-convert_mae_se_assay_to_dt.R:206]: redundant `gDRutils::` namespacing and in other places

File: R/convert_mae_se_assay_to_dt.R (hunk 1 above)
Comment (j-smola) [reply R/convert_mae_se_assay_to_dt.R]: Done.

File: tests/testthat/test-convert_mae_se_assay_to_dt.R [hunk 3]
```diff
@@ -197,3 +197,98 @@ test_that("convert_mae_assay_to_dt works as expected", {
...
+                    "x_inf", "x_0", "h", "r2", "x_sd_avg", "fit_type") %in% names(dt3)))
+  dt4 <- convert_se_assay_to_custom_dt(se, assay_name = "Averaged")
+  checkmate::expect_data_table(dt2, min.rows = 2, min.cols = 2)
+  expect_true(
+    all(c("GR value", "Relative Viability", "Std GR value", "Std Relative Viability") %in% names(dt4)))
+  dt5 <- convert_se_assay_to_custom_dt(se, assay_name = "Averaged", output_table = "Metrics")
+  expect_true(identical(dt4, dt5))
+  
+  se2 <-
+    gDRutils::get_synthetic_data("finalMAE_combo_matrix")[[1]]
```
Comment (bczech) [inline tests/testthat/test-convert_mae_se_assay_to_dt.R:229]: Redundant namespacing here and other places in this script `gDRutils::`

**Review by gladkia — APPROVED** [review]

File: tests/testthat/test-utils.R [hunk 4]
```diff
@@ -465,3 +465,96 @@ test_that("get_additional_variables works as expected", {
   expect_equal(get_additional_variables(unlist(rdata1)), NULL)
   
 })
+
+
+test_that("convert_se_assay_to_custom_dt works fine", {
+  json_path <- system.file(package = "gDRcomponents", "settings.json")
+  s <- get_settings_from_json(json_path = json_path)
+  
+  se <- gDRutils::get_synthetic_data("finalMAE_small")[[1]]
```
Comment (bczech) [inline tests/testthat/test-utils.R:474]: still here `gDRutils::` and below

File: tests/testthat/test-convert_mae_se_assay_to_dt.R [hunk 5]
```diff
@@ -197,3 +197,97 @@ test_that("convert_mae_assay_to_dt works as expected", {
...
+    all(c("GR value", "Relative Viability", "Std GR value", "Std Relative Viability") %in% names(dt4)))
+  dt5 <- convert_se_assay_to_custom_dt(se, assay_name = "Averaged", output_table = "Metrics")
+  expect_true(identical(dt4, dt5))
+  
+  se2 <- get_synthetic_data("finalMAE_combo_matrix")[[1]]
+  dt6 <- convert_se_assay_to_custom_dt(se2, assay_name = "Metrics")
+  checkmate::expect_data_table(dt6, min.rows = 2, min.cols = 2)
+  expect_true(all(s$METRIC_WISH_LIST %in% names(dt6)))
+  dt7 <-
+    convert_se_assay_to_custom_dt(se2, assay_name = gDRutils::get_combo_assay_names()[1])
```
Comment (bczech) [inline tests/testthat/test-convert_mae_se_assay_to_dt.R:233]: here

File: tests/testthat/test-utils.R [hunk 6]
```diff
@@ -465,3 +465,96 @@ test_that("get_additional_variables works as expected", {
...
+                    "x_inf", "x_0", "h", "r2", "x_sd_avg", "fit_type") %in% names(dt3)))
+  dt4 <- convert_se_assay_to_custom_dt(se, assay_name = "Averaged")
+  checkmate::expect_data_table(dt2, min.rows = 2, min.cols = 2)
+  expect_true(
+    all(c("GR value", "Relative Viability", "Std GR value", "Std Relative Viability") %in% names(dt4)))
+  dt5 <- convert_se_assay_to_custom_dt(se, assay_name = "Averaged", output_table = "Metrics")
+  expect_true(identical(dt4, dt5))
+  
+  se2 <-
+    gDRutils::get_synthetic_data("finalMAE_combo_matrix")[[1]]
```
Comment (bczech) [inline tests/testthat/test-utils.R:497]: here

File: tests/testthat/test-utils.R [hunk 7]
```diff
@@ -465,3 +465,96 @@ test_that("get_additional_variables works as expected", {
...
+  dt6 <- convert_se_assay_to_custom_dt(se2, assay_name = "Metrics")
+  checkmate::expect_data_table(dt6, min.rows = 2, min.cols = 2)
+  expect_true(all(s$METRIC_WISH_LIST %in% names(dt6)))
+  dt7 <-
+    convert_se_assay_to_custom_dt(se2, assay_name = gDRutils::get_combo_assay_names()[1])
```
Comment (bczech) [inline tests/testthat/test-utils.R:502]: here

File: tests/testthat/test-utils.R [hunk 8]
```diff
@@ -465,3 +465,96 @@ test_that("get_additional_variables works as expected", {
...
+  checkmate::expect_data_table(dt7, min.rows = 2, min.cols = 2)
+  expect_true(all(names(gDRutils::get_combo_excess_field_names()) %in% names(dt7)))
```
Comment (bczech) [inline tests/testthat/test-utils.R:504]: here

File: tests/testthat/test-convert_mae_se_assay_to_dt.R (hunk 3 above)
Comment (j-smola) [reply tests/testthat/test-convert_mae_se_assay_to_dt.R]: ✔️

**Review by bczech — APPROVED** [review]

File: tests/testthat/test-convert_mae_se_assay_to_dt.R (hunk 3 above)
Comment (j-smola) [reply tests/testthat/test-convert_mae_se_assay_to_dt.R]: ✔️

File: tests/testthat/test-utils.R (hunk 4 above)
Comment (j-smola) [reply tests/testthat/test-utils.R]: ✔️

File: tests/testthat/test-convert_mae_se_assay_to_dt.R (hunk 5 above)
Comment (j-smola) [reply tests/testthat/test-convert_mae_se_assay_to_dt.R]: ✔️

File: tests/testthat/test-utils.R (hunk 6 above)
Comment (j-smola) [reply tests/testthat/test-utils.R]: ✔️

File: tests/testthat/test-utils.R (hunk 7 above)
Comment (j-smola) [reply tests/testthat/test-utils.R]: ✔️

File: tests/testthat/test-utils.R (hunk 8 above)
Comment (j-smola) [reply tests/testthat/test-utils.R]: ✔️

**Pull request merged**